### PR TITLE
[IMP] users: review of my profile view

### PR DESCRIPTION
**Pull request by kbapt**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/hr_attendance/models/res_users.py [hunk 1]
```diff
@@ -8,7 +8,10 @@ class User(models.Model):
     _inherit = ['res.users']
 
     hours_last_month = fields.Float(related='employee_id.hours_last_month')
+    hours_last_month_display = fields.Char(related='employee_id.hours_last_month_display')
     attendance_state = fields.Selection(related='employee_id.attendance_state')
+    last_check_in = fields.Datetime(related='employee_id.last_attendance_id.check_in', store=True)
```
Comment (RomainLibert) [inline addons/hr_attendance/models/res_users.py]: Why is it stored here too, is it really necessary?

File: addons/hr_attendance/models/res_users.py [hunk 2]
```diff
@@ -8,7 +8,10 @@ class User(models.Model):
...
+    last_check_out = fields.Datetime(related='employee_id.last_attendance_id.check_out', store=True)
```
Comment (RomainLibert) [inline addons/hr_attendance/models/res_users.py]: Why is it stored here too, is it really necessary?

File: addons/hr_attendance/static/src/js/time_widget.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+            if (!(val && '_isAMomentObject' in val && val._isAMomentObject)) {
+                return;
+            }
+            return val.fromNow(true);
+        },
+    });
+
+    registry.add('relative_time', RelativeTime);
+
+    return { RelativeTime: RelativeTime }
```
Comment (RomainLibert) [inline addons/hr_attendance/static/src/js/time_widget.js]: There is only one widget defined here, you can simply return it

File: addons/hr_attendance/static/src/js/time_widget.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+odoo.define('hr_attendance.widget', function (require) {
+    "use strict";
+
+    var basic_fields = require('web.basic_fields');
```
Comment (RomainLibert) [inline addons/hr_attendance/static/src/js/time_widget.js]: Don't import basic_fields directly

File: addons/hr_attendance/static/src/js/time_widget.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    var registry = require('web.field_registry');
```
Comment (RomainLibert) [inline addons/hr_attendance/static/src/js/time_widget.js]: Call it field_registry

File: addons/hr_attendance/static/src/js/time_widget.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+    var RelativeTime = basic_fields.FieldDateTime.extend({
```
Comment (RomainLibert) [inline addons/hr_attendance/static/src/js/time_widget.js]: Use field_registry to get the FieldDateTime in order to extend it

File: addons/hr_attendance/static/src/js/time_widget.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+        _formatValue: function (val) {
+            if (!(val && '_isAMomentObject' in val && val._isAMomentObject)) {
```
Comment (RomainLibert) [inline addons/hr_attendance/static/src/js/time_widget.js]: Doesn't this do the same?
``` Javascript
if (!(val && val._isAMomentObject))
```

File: addons/hr_recruitment/models/hr_recruitment.py [hunk 8]
```diff
@@ -458,6 +458,9 @@ def create_employee_from_applicant(self):
                     'work_phone': applicant.department_id and applicant.department_id.company_id
                             and applicant.department_id.company_id.phone or False})
                 applicant.write({'emp_id': employee.id})
+                applicant.message_post(
+                    body=_('Applicant hired<br><ul><li>Employee: <a href="#" data-oe-id="%s" data-oe-model="hr.employee">%s</a></li></ul>') % (applicant.emp_id.id, applicant.emp_id.name),
```
Comment (tivisse) [inline addons/hr_recruitment/models/hr_recruitment.py]: Make a qweb template instead. This is nightmare for the translations.

File: addons/hr_recruitment/models/hr_recruitment.py [hunk 9]
```diff
@@ -150,10 +150,10 @@ def _default_company_id(self):
     day_close = fields.Float(compute='_compute_day', string="Days to Close")
     delay_close = fields.Float(compute="_compute_day", string='Delay to Close', readonly=True, group_operator="avg", help="Number of days to close", store=True)
     color = fields.Integer("Color Index", default=0)
-    emp_id = fields.Many2one('hr.employee', string="Employee", tracking=True, help="Employee linked to the applicant.")
+    emp_id = fields.Many2one('hr.employee', string="Employee", help="Employee linked to the applicant.")
     user_email = fields.Char(related='user_id.email', type="char", string="User Email", readonly=True)
     attachment_number = fields.Integer(compute='_get_attachment_number', string="Number of Attachments")
-    employee_name = fields.Char(related='emp_id.name', string="Employee Name", readonly=False)
+    employee_name = fields.Char(related='emp_id.name', string="Employee Name", readonly=False, tracking=False)
```
Comment (tivisse) [inline addons/hr_recruitment/models/hr_recruitment.py]: tracking=False should be implicit.

File: addons/web/static/src/js/fields/field_utils.js [hunk 10]
```diff
@@ -106,7 +106,7 @@ function formatChar(value, field, options) {
  * @returns {string}
  */
 function formatDate(value, field, options) {
-    if (value === false) {
+    if (value === false || isNaN(value)) {
```
Comment (tivisse) [inline addons/web/static/src/js/fields/field_utils.js]: To validate with someone from the framework JS team.

File: addons/hr/views/hr_employee_public_views.xml [hunk 11]
```diff
@@ -8,6 +8,15 @@
             <field name="arch" type="xml">
                 <search string="Employees">
                     <field name="name" string="Employees" filter_domain="['|',('work_email','ilike',self),('name','ilike',self)]"/>
+                    <field name="job_title" string="Job Title" filter_domain="[('job_title','ilike',self)]"/>
```
[the remark this answers is not in the document]
Comment (tivisse) [reply addons/hr/views/hr_employee_public_views.xml]: filter_domain should be used to make something more specific than a simple search based on the record name. 

If you don't specify a filter_domain here, you would have the same result.

File: addons/hr_recruitment/models/hr_recruitment.py (hunk 9 above)
Comment (kba-odoo) [reply addons/hr_recruitment/models/hr_recruitment.py]: I had to explicitly untrack it, it was tracked from hr.employee and showing twice as I'm posting it as well with a link to the employee

**Comment by tivisse** [comment]
@robodoo r+

**Comment by tivisse** [comment]
@robodoo r+

**Comment by tivisse** [comment]
@robodoo r+

**Comment by tivisse** [comment]
@robodoo r+

File: addons/hr_recruitment/models/hr_recruitment.py [hunk 12]
```diff
@@ -457,6 +458,10 @@ def create_employee_from_applicant(self):
                     'work_phone': applicant.department_id and applicant.department_id.company_id
                             and applicant.department_id.company_id.phone or False})
                 applicant.write({'emp_id': employee.id})
+                applicant.message_post_with_view(
+                    'hr_recruitment.applicant_hired_template',
+                    values={'applicant': applicant},
+                    subtype_id=self.env.ref("hr_recruitment.mt_applicant_hired").id)
```
Comment (tivisse) [inline addons/hr_recruitment/models/hr_recruitment.py]: subtype is not a valid argument, leading to a warning on runbot.

The correct way to handle this is

subtype_id=self.env.ref("hr_recruitment.mt_applicant_hired").id
message_post_with_view has not the same signature than message_post by the way.

**Comment by tivisse** [comment]
@robodoo r+

**Comment by frlanxpertiz** [comment]
this change has removed the field medic_exam from the employee form even though there is no mention of it in the commit message. Possibly a mistake? If that was voluntary shouldn't the field be removed from the model as it's not used anywhere?

**Comment by frlanxpertiz** [comment]
@tivisse @kba-odoo

**Comment by tivisse** [comment]
Hi @frlanxpertiz 

Thanks for the feedback. Indeed this field could be removed. 

I'm creating an internal task to manage this.

Have a nice day,

Yannick.

**Pull request closed without merging**